frontend/src/components/SystemSubmitDrawer/index.tsx

```diff
@@ -130,15 +162,33 @@ export function SystemSubmitDrawer(props: Props) {
   const [useCustomDataset, setUseCustomDataset] = useState(false);
   const [otherSourceLang, setOtherSourceLang] = useState(false);
   const [otherTargetLang, setOtherTargetLang] = useState(false);
-
   const [form] = useForm<FormData>();
-  const { systemToEdit, onClose, ...rest } = props;
+  const [codeGenFields, setCodeGenFields] = useState(
```

Comment:
`codeGenFields` can be derived from `form.getFieldsValue()` and `useCustomDataset` so I think it does not need to be a state. Maintaining this state seems to be relatively complicated (we need to call `parseFormFields` on different values). I think it will make things easier if we compute `codeGenFields` on the fly whenever we need to rerender `<ClientCodeDisplay />`.

This would create an issue though. Because of the way `<Form />` behaves, a value change of `<Form.Item />` won't change `form`. So, it won't trigger a rerender of `<ClientCodeDisplay />` when a value is changed. I think an easy solution would be to write something like this:
```
  const [, updateState] = React.useState<boolean>();
  const forceUpdate = React.useCallback(() => updateState((s) => !s), []);
```
And you can call `forceUpdate()` whenever you need to rerender `<ClientCodeDisplay />` (e.g. whenever `form.getFieldsValue()` changes, essentially whenever `onValuesChange()` is invoked). This is not the most efficient solution because it rerenders the entire `<SystemSubmitDrawer />` but I think it would make the code easier to understand. A more efficient solution would be to wrap `<ClientCodeDisplay />` with another dummy component. And trigger rerender for that dummy component whenever we need to rerender `<ClientCodeDisplay />`.

Reply:
I removed that state and implemented the easy `forceUpdate()` solution for now.